scripts/corr2_rx_bf.py

```diff
@@ -0,0 +1,261 @@
+    parser.add_argument('--config', dest='config', action='store', default='',
+                        help='corr2 config file')
+    parser.add_argument('--file', dest='writefile', action='store_true', default=False,
+                        help='Write an H5 file.')
+    parser.add_argument('--filename', dest='filename', action='store', default='',
+                        help='Use a specific filename, otherwise use UNIX time. Implies --file.')
+    parser.add_argument('--plot_baselines', dest='plotbaselines', action='store', default='', type=str,
+                        help='Plot one or more baselines, comma-seperated list of integers.')
+    parser.add_argument('--plot_channels', dest='plotchannels', action='store', default='-1,-1', type=str,
+                        help='a start,end tuple, -1 means 0,n_chans respectively')
+    parser.add_argument('--item', dest='items', action='append', default=[],   
+                        help='spead item to output value to log as we receive it')
+    parser.add_argument('--ion', dest='ion', action='store_true', default=False,
+                        help='Interactive mode plotting.')
+    parser.add_argument('--log', dest='log', action='store_true', default=False,
+                        help='Logarithmic y axis.')
+    parser.add_argument(
+        '--no_auto', dest='noauto', action='store_true', default=False,
+        help='Do not scale the data by the number of accumulations.')
+    parser.add_argument(
+        '--loglevel', dest='log_level', action='store', default='DEBUG',
+        help='log level to use, default INFO, options INFO, DEBUG, ERROR')
+    parser.add_argument(
+        '--speadloglevel', dest='spead_log_level', action='store',
+        default='ERROR',
+        help='log level to use in spead receiver, default INFO, '
+             'options INFO, DEBUG, ERROR')
+    args = parser.parse_args()
+
+    if 'CORR2INI' in os.environ.keys() and args.config == '':
+        args.config = os.environ['CORR2INI']
+
+    # load the rx port and sd info from the config file
+    config = utils.parse_ini_file(args.config)
+    data_port = int(config['xengine']['output_destination_port'])
+    n_chans = int(config['fengine']['n_chans'])
+    print 'Loaded instrument info from config file:\n\t%s' % args.config
+
+    if args.items:
+        items = args.items
+        print 'Tracking:'
+        for item in items:
+            print '\t  * %s' % item
+    else:
+        print 'Not tracking any items.'
+
+print 'Initialising SPEAD transports for data:'
+print '\tData reception on port', data_port
+if filename is not None:
+    print '\tStoring to file %s' % filename
+else:
+    print '\tNot saving to disk.'
+
+quit_event = threading.Event()
+memory_error_event = threading.Event()
+plotqueue = Queue.Queue()
+got_data_event = threading.Event()
+spead_queue = Queue.Queue()
+
+
+if args.ion:
+    pyplot.ion()
+
+
+def plot():
+    if got_data_event.is_set():
```

Comment:
If using the above size 1 Queue suggestion, de-indent the body of the try: stanza twice and get rid of the if statment, and replace `plotqueue.get_nowait()` with `plotqueue.get()`. Or keep the try statement if you want to add a timeout to `get()`, prolly not a bad idea since IIRC get() w/o a timeout swallows ctrl-C